.github/workflows/publish.yml

```diff
@@ -14,14 +14,14 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v2
       with:
-        python-version: '3.8'
+        python-version: '3.x'
     - name: Install dependencies and build
       run: |
-        python -m pip install --upgrade pip wheel
-        pip install twine
-        python setup.py sdist bdist_wheel
-    - name: Publish to PyPI
-      uses: pypa/gh-action-pypi-publish@master
-      with:
-        user: __token__
-        password: ${{ secrets.PYPI_API_TOKEN }}
+        python -m pip install --upgrade pip
+        pip install poetry poetry-dynamic-versioning twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```

Comment:
We can use TWINE_USERNAME as `__token__` like below
https://github.com/m3dev/gokart/blob/master/.github/workflows/publish.yml#L26-L27

ref: 
https://pypi.org/help/#apitoken